src/Microsoft.TestPlatform.ObjectModel/DataCollector/IDataCollectorAttachments.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Interface for data collectors add-ins that choose to handle attachment(s) generated
+    /// </summary>
+    public interface IDataCollectorAttachments
+    {
+        /// <summary>
+        /// Gets the attachment set after Test Run Session
+        /// </summary>
+        /// <returns>Gets the attachment set after Test Run Session</returns>
+        ICollection<AttachmentSet> HandleDataCollectionAttachmentSets(ICollection<AttachmentSet> dataCollectionAttachments);
+
+        /// <summary>
+        /// Gets the attachment Uri, which is handled by current Collector
+        /// </summary>
+        Uri GetAttachmentUri();
```

Comment:
This interface is supposed handle attachments from a specific data collector extension, not the individual attachments, consider renaming this method to `GetDataCollectorUri` or `GetExtensionUri`